### ISSUE TITLE
Feature Request: Batch Synchronization from .srt Files with Corrected Timestamps

### ISSUE THREAD

**Issue by babywin** [denizsafak/AutoSubSync#6]
Hi @denizsafak,  

Thank you for implementing synchronization with alass—it works wonderfully! I truly appreciate your efforts.  

I would like to request a feature that allows batch mode synchronization starting from .srt files with corrected timestamps. As far as I can tell, the current implementation only supports synchronization from a video file.  

Thank you once again for your amazing work!

**Comment by denizsafak** [comment]
You can add `reference subtitle - subtitle` pairs in batch mode. Just try to add subtitles and it will ask you how you want to add them. You can add them as subtitles with [no video] or reference subtitles with [no subtitle] input. After adding, just pair them with another subtitles manually.

**Comment by denizsafak** [comment]
I also updated the script. Please install v2.7 again.

**Comment by babywin** [comment]
I see that it's possible to pair files manually, one by one. However, I wonder if there's a way to implement an automatic pairing feature for subtitles. For example, could the "drop here" box be divided into two sections? The upper section could allow users to "drop all video/subtitle files," while the lower section could specify "drop all unsynced subtitle files," similar to the layout in Normal Mode. 

For instance, if the upper box contains:  
- Show S01E01.srt  
- Show S01E02.srt
or
-Show S01E01.en.srt  
-1x02 Episode Title.srt

And the lower box contains:  
- Show S01E01.srt  
- Show S01E02.srt  

The system could automatically pair the files with matching names, just as it does when video and subtitle files share the same name.

I also have a question about the Batch pairing process. When pairing videos and subtitles, is there a way to force the search for embedded `.srt` files within the video file? I've noticed that, in some cases, the tool skips this step and instead synchronizes using the video's audio track, even when subtitles are present (maybe it's just me).

Lastly, is it possible to configure the sync process to prioritize subtitles labeled as "non-hearing impaired" (NO-HI) for videos that include both regular and HI subtitles?  

Thank you once again for this outstanding tool!

**Comment by denizsafak** [comment]
Pairing subtitles that has similar filenames is possible, but their filenames should be in specific formats. Otherwise it won't work. There are so many combinations and factors to check for pairing them correctly and it is hard to achieve. Of course it will work for some, but won't work all the time.

About forcing the search for embedded .srt files, [ffsubsync ](https://github.com/smacke/ffsubsync) checks the subtitles in the video file. If it can't find any _usable_ subtitle in the video, it uses the audio track for syncing. Sometimes I face the same issue as you, but it's not something that I can fix. You need to report an issue in [ffsubsync](https://github.com/smacke/ffsubsync) for that.

Lastly, for prioritizing subtitles, [ffsubsync](https://github.com/smacke/ffsubsync) selects the subtitles, and again, sadly, there is nothing I can do.

**Comment by babywin** [comment]
I believe that implementing batch pairing for subtitles with identical filenames would be an excellent addition to AutoSubSync. 

Regarding the possibility of forcing the search for embedded `.srt` files, I was considering `alass` for this functionality rather  `ffsubsync`.

Finally, I don't want to take up too much of your time, @denizsafak. Thank you once again for creating such an incredibly useful tool!

**Comment by denizsafak** [comment]
Thank you 🙂 Maybe I'll try to add them in the next release. Have a nice day 😋

**Comment by denizsafak** [comment]
Added the feature that you mentioned about extracting subtitles from videos for alass, in the [latest](https://github.com/denizsafak/AutoSubSync/releases/tag/v2.9) release. It turns out I was able to do that. About prioritizing (NO-HI) subtitles, you probably won't need that because it scans all extracted subtitles and compares them with your subtitle then selects the best one automatically. If you face any issues, feel free to open another issue. Next step: Adding reference subtitle/subtitle pairing mode 😁
